### PR TITLE
Draft: add GuiCommandBase class to serve as the parent class of all DraftTools

### PR DESCRIPTION
**Pull request by vocx-fc**
Prepare auxiliary directories to restructure all graphical tools for objects, viewproviders, gui commands, and taskpanels.

This pull request follows from the topic discussed in this thread, [[Discussion] Splitting Draft tools into their own modules](https://forum.freecadweb.org/viewtopic.php?f=23&t=38593), particularly from this [post](https://forum.freecadweb.org/viewtopic.php?p=341298#p341298).

Pull request #2429 was a first attempt which shows how it could be done tentatively, but the present request is a formal attempt to re-structure the code.

We will start by establishing a base class for all GUI commands, which in the original code is `DraftTools.DraftTool`. This class is heavily tied together with the `DraftGui.DraftToolbar` class which sets up the graphical [task panel](https://www.freecadweb.org/wiki/Task_panel) of each command in Draft. We don't want to have a task panel defined manually, we prefer to use the `.ui` files created by QtCreator. So, we will use a new class `GuiCommandBase` to untangle and re-code the relevant commands, and provide individual `.ui` files.

In a future pull request we will provide new commands that currently don't exist: [Draft_PolarArray](https://www.freecadweb.org/wiki/Draft_PolarArray) and [Draft_CircularArray](https://www.freecadweb.org/wiki/Draft_CircularArray). These will be new in 0.19, and will show how to structure the code for the future. We will also try to re-structure the old commands, little by little.

---

- [x] Branch rebased on latest master `git pull --rebase upstream master`
- [x] Unit tests confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [x] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)
- [ ] Commit message includes `issue #<id>` or `fixes #<id>` where `<id>` is the [associated MantisBT](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) issue id if one exists

### CODE REVIEW

**Comment by yorikvanhavre** [comment]
Great! Remember that all file changes used to need a corresponding PR on the debian salsa "freecad-daily" repo. But I don't see the file anymre... @kkremitzki is this still needed?

**Comment by kkremitzki** [comment]
No, that's not needed anymore.

**Comment by vocx-fc** [comment]
Kremitzki has discussed this in the forum, that there is now a new way to pick up the files in the build directory for packaging. As far as I could tell, it was just asterisk expansion.

Something like this
```
freecad-daily/Mod/*
freecad-daily/Ext/*
```

So, hopefully this works okay, but if it doesn't do let me know.

This was already tested in #2727, where I moved a lot of functions to individual modules and they were picked up okay by the daily builds, so I expect this to work okay as well.

**Comment by yorikvanhavre** [comment]
All right then! Sorry, getting hard to follow everything on the forum

**Comment by vocx-fc** [comment]
By the way, I just discovered it doesn't work perfectly. See my notes on #2829 and in the forum post indicated there. I need to fix the CMakeLists file to get things installed properly in Travis and possibly other systems like the freecad-daily.
